app/services/google/scrape_service.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Google
+  class ScrapeService
+    def initialize(keyword_id:, search_service: Google::SearchService.new)
+      @keyword_id = keyword_id
+      @search_service = search_service
+    end
+
+    def call!
+      keyword = Keyword.find_by id: keyword_id
+      raise_keyword_not_found unless keyword
```

Comment:
why not just use the `Keyword.find(keyword_id)` like previous version and we catch on the RecordNotFound exception like before 🤔

Reply:
oh I think this makes it clearer where the error came from. Try to avoid potential confusion

Reply:
no need to do that @Goose97 as in Rails, we know that the `find(xx)` will raise the NotFound exception, so we just catch it.

Reply:
Thanks for the note. Fixed in 736b1d120535fc0713ebd5fa47eb230a634031b8